api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/BaseCodingTranslator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import javax.annotation.Nonnull;
+
+import lombok.Setter;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.ICoding;
+import org.openmrs.Concept;
+import org.openmrs.module.fhir2.api.translators.ConceptTranslator;
+import org.openmrs.util.OpenmrsUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Setter
+public abstract class BaseCodingTranslator {
+	
+	@Autowired
+	protected ConceptTranslator conceptTranslator;
+	
+	/**
+	 * Base implementation of conversion between a FHIR Coding interface and OpenMRS Concept
+	 * representation.
+	 */
+	protected Concept fhirCodingToOpenmrsConcept(@Nonnull ICoding coding) {
+		if (coding.getCode() != null) {
+			CodeableConcept codeableConcept = new CodeableConcept();
+			codeableConcept.addCoding(new Coding(coding.getSystem(), coding.getCode(), coding.getDisplay()));
+			return conceptTranslator.toOpenmrsType(codeableConcept);
+		}
+		return null;
+	}
+	
+	/**
+	 * @return the coding on the CodeableConcept with the given system, or null if none found.
+	 */
+	public static Coding getCodingForSystem(CodeableConcept codeableConcept, String system) {
```

Comment:
initial thought was to add this utility function here since its related to coding translation, but we could also move it over to FhirUtils instead

Reply:
I think it's fine to keep it here for now... shouldn't be too disruptive to move it later. Should this method be `protected` though?

Reply:
hmm not sure to be honest. the only benefit of a protected static method in this case would be so we restrict its use to within its package. is that something we want to do? 

i think i'd prefer to keep it public